analysis_file/analysis_file.py

```diff
@@ -0,0 +1,154 @@
+        description='Generates files for analysis from the cleaned and coded '
+        'show and survey responses')
+    parser.add_argument(
+        'user', help='User launching this program')
+    parser.add_argument(
+        'json_input_path', metavar='json-input-path',
+        help='Path to a coded JSON file, containing a list of serialized '
+        'TracedData objects')
+    parser.add_argument(
+        'json_output_path', metavar='json-output-path',
+        help='Path to a JSON file to write serialized TracedData items to '
+        'after modification by this pipeline stage')
+    parser.add_argument(
+        'csv_by_message_output_path', metavar='csv-by-message-output-path',
+        help='Analysis dataset where messages are the unit for analysis (i.e. '
+        'one message per row)')
+    parser.add_argument(
+        'csv_by_individual_output_path', metavar='csv-by-individual-output-path',
+        help='Analysis dataset where respondents are the unit for analysis '
+        '(i.e. one respondent per row, with all their messages joined into a '
+        'single cell).')
+
+    args = parser.parse_args()
+    user = args.user
+    data_input_path = args.json_input_path
+    json_output_path = args.json_output_path
+    csv_by_message_output_path = args.csv_by_message_output_path
+    csv_by_individual_output_path = args.csv_by_individual_output_path
+
+    # Serializer is currently overflowing
+    # TODO: Investigate/address the cause of this.
+    sys.setrecursionlimit(10000)
+
+    demog_keys = [
+        'gender',
+        'gender_raw',
+        'age',
+        'age_raw',
+        'clan_identity',
+        'clan_identity_raw']
+
+    survey_keys = [
+        'needs_met_yesno',
+        'needs_met_raw',
+        'cash_modality_yesno',
+        'cash_modality_raw',
+        'community_priorities_raw',
+        'inclusion_yesno',
+        'inclusion_raw']
+    
+    scope_keys = [
+        'scope_district',
+        'HH_size']
+
+    key_map = {
+        'UID': 'avf_phone_id',
+        'needs_met_raw': 'Needs_Met_Yesno (Text) - emergency_maap_new_pdm',
+        'cash_modality_raw': 'Cash_Modality_Yesno (Text) - emergency_maap_new_pdm',
+        'community_priorities_raw': 'Community_Priorities (Text) - emergency_maap_new_pdm',
+        'inclusion_raw': 'Inclusion_Yes_No (Text) - emergency_maap_new_pdm',
+        
+        'gender': 'gender_coded',
+        'gender_raw': 'Gender (Text) - emergency_maap_new_demogs',
+        'age': 'age_coded',
+        'age_raw': 'Age (Text) - emergency_maap_new_demogs',
+        'clan_identity': 'clan_coded',
+        'clan_identity_raw': 'Clan (Text) - emergency_maap_new_demogs',
+        
+        'scope_district': 'District',
+        'household_size': 'Household Size'}
+
+    # Load cleaned and coded message/survey data
+    with open(data_input_path, 'r') as f:
+        list_td = TracedDataJsonIO.import_json_to_traced_data_iterable(f)
+
+    keys_to_matrix = [
```

Comment:
Are these being set correctly after folding? Notify me if not

Reply:
That's the thing, so far we've only had single responses to so I've not be able to check it's working well.

Reply:
Need an answer to this before I can approve

Reply:
Seem to be working fine.